### ISSUE TITLE
 Support BSIP-0029 - Changed operation to update issuer of an asset

### ISSUE THREAD

**Issue by startailcoon** [bitshares/bitshares-ui#1596]
# BSIP-0029 Changes

BSIP-0029 makes a change in the core that moves the way you change issuer of an asset.
[See BSIP-0029](https://github.com/bitshares/bsips/blob/master/bsip-0029.md).

### Current Implementation
Currently, any asset can be updated with `asset_update_operation`. This operation contains an optional `new_issuer` that changes the issuer of the asset.

### New Implementation
`new_issuer` no longer exists in `asset_update_operation`, but rather in its own operation `asset_update_issuer_operation`.

**Comment by abitmore** [comment]
Personally I prefer this one: Add a "Change Asset Issuer/Owner" button on the Update Owner section of account/<account>/update-asset/<asset>.

**Comment by happyconcepts** [comment]
> everyone can burn every UIA. 

This does not sound right. Please show me where this is even intended.


Why not simply update the existing button in the existing UI to reference the new operation?

No functionality is changing, so why create UX and other work here?

**Comment by abitmore** [comment]
@happyconcepts 
1. user story: if someone sent me some shitcoin and I don't want it, I burn it.
2. functionality IS changing.

**Comment by startailcoon** [comment]
Just as @abitmore sais, this is labeled UX because we have a change in functionality. UX label doesn't mean a massive workload, but rather that we need to find a good standpoint on it for the issue.

The change in functionality is that the asset owner change is no longer inside the same operation as the rest of the asset properties. We then need to find a good stance on where this function should be.

**Comment by happyconcepts** [comment]
Both of you need to start taking me seriously, please. 

This bsip is to make it harder for any attacker to change the issuer of a coin.

@xeroc

**Comment by happyconcepts** [comment]
@abitmore with respect ...

> 1. user story: if someone sent me some
> shitcoin and I don't want it, I burn it.
> 2. functionality IS changing.

1. This issue is related to **changing the issuer of a coin**  so ... wtf does your shitcoin burning story have to do with it?

If @dan over at eos wakes up tomorrow and decides that BTS is a shitcoin (example only, ) you are going to let him burn all his BTS and wreck the system huh.

2. So disappointed that you think I cannot read the bsip in my own mother tongue and understand it.

How can we open your mind or ego a little bit more so there is some room for other people in bitshares? 

Once you stop fighting every bit of my effort ... only then can constructive work begin to get done.

A shitty system can only produce shitty coin, so if there is too much shitty coin in your opinion....let's fix the system. Not make it easier for you to label everything as shit.

**Comment by clockworkgr** [comment]
With respect to both of you, I believe there was a misunderstanding here.

@abitmore mentioned that "everyone can burn any UIA" as a side-note, NOT related to this issue simply to point out that BURN ASSET should not be in that page because it's not only the issuer that can burn but everyone else as well.

**Comment by happyconcepts** [comment]
When the intent of Fabien's bsip 0029 is so clear ...

And this work issue to implement it is so conflated, and padded with UX for no reason, and simply misdirected ... 

Was the Burn Asset button sensibly placed  for a sensible reason in the ASSET Management screen? Yes, because only the issuer should be able to control this function.

Why is your attitude to continue to take away the issuer's control over their creation (e.g. the UIA) ?

**Comment by clockworkgr** [comment]
II don't know if there are personal differences here , but as an outsider, it's not that big of a deal. Just 2 things:

a) The change does need UX, because it used to be the case that you clicked on UPDATE ASSET and could update everything including issuer. This is now a different operation, therefore it needs somewhere to go so the issuer can execute it. That is all.

b) as a sidenote due to the screenshot that was posted, abit mentioned how that button should be moved because it's in the wrong place considering the chain's functionality. I do not think it was placed there to "hide" it but by wrong assumption. 
If you have issues with the ACTUAL functionality and believe only the issuer should be able to burn the asset then that is a discussion for -core. Not -ui.

**Comment by svk31** [comment]
You're misunderstanding the burn comments by abit. Anyone can burn an asset *they have in their wallet* is what he means. I can burn the open.btc I have in my wallet for example, but openledger cannot, that's why this burn option doesn't really belong in this page.

**Comment by happyconcepts** [comment]
@svk31 ty for clarifying for me about the burn
@abitmore apologies for my misunderstanding on the burn.

Now, back to this issue about changing issuers of coins..please.

User story: my bot now issues [my coin SPARK](http://cryptofresh.com/a/SPARK) and is ready to issue coins for other coin owners so .... this bsip seems highly relevant.

So when George and Maxine over at bitSPARK hire me and my bot to issue their  bitSPARK coin or ZEPHYR coin to their users, they provide me the key to issue their  coins.

This bsip simply says that another key would be required in order to change the issuer/owner of their coins, in case my bot is hacked for example and the attackers gain access to the key I have for bitSPARK. 

In a subtle yet very important way, this bsip is about making it harder to change issuer, not easier. It's about security.

I see it as simple and straightforward as reviewing the UI code for where to add the requirement of the additional key for the specific `asset_update_operation` transaction. It's all literally behind the screens.

**Comment by abitmore** [comment]
@happyconcepts "issuing" coins doesn't need owner key. I think it's not related to this issue. Only changing an asset's issuer need owner key, and need to be done with a new operation `asset_update_issuer_operation` after the scheduled hard fork time (July 19). In addition, after that time `asset_update_operation` can not be used to change assets' issuer.

When talking about your bot, if it's designed to issue coins automatically, most likely your active key is on the server. Before the hard fork, if your server got hacked, the hacker can change your assets' issuers to himself with the active key, so you'll completely lose control over the assets. After the hard fork, the hacker can't change the issuers if your owner key is not on the server. That's it.

By the way, I don't believe your bot is running clicking buttons on UI. Why an UI change is relevant to the bot?

**Comment by clockworkgr** [comment]
@happyconcepts  considering the use case you mentioned, you might also want to have a look at https://github.com/bitshares/bitshares-core/issues/1061 which I'm hoping will turn into a BSIP.

Will be happy to hear any comments you might have.

**Comment by happyconcepts** [comment]
> After the hard fork, the hacker can't change the issuers if your owner key is not on the server. That's it.

But the owner key is still always in my light client, since the light client UI is used in my business.

So I am concerned with the security of the owner key in the UI in general, and specifically more so when you propose adding a change asset issuer feature to the UI.

@abitmore where please in the current bitshares UI to change the owner of a UIA?

**Comment by svk31** [comment]
It's in the update asset screen now, but that doesn't work anymore now that the HF is live.

**Comment by clockworkgr** [comment]
@svk31 HF is not live till july...so still working till then

**Comment by abitmore** [comment]
@happyconcepts yes you do need to import keys to sign certain operations. For better security, you follow better practice, for example use multi-sig, offline signing and etc. Every change comes with pros and cons, if you have a question or demand about the back end (chain logic change but not only the UI), please create an issue or even BSIP in https://github.com/bitshares/bsips repository. It's up to stake holders to decide whether to change and how to change, via voting. Here is not the correct place for this discussion.

**Comment by startailcoon** [comment]
Well, if we can get back on track with this issue, this is what we need to work on.

# UX
[ ] Placement of `Asset Issuer Change`

It has been proposed to add this button within the Asset Update section. This has to clearly visible that it requires the action within that page, and isn't included within the normal update.

Another possible way to this is to make use of sending multiple transactions at the update. If the issuer is changed, we issue a second command to do this as well.

# UI
Depending on the way we choose to update the UX.

## Additional Task could be
[ ] Move the `Burn Asset` to another, more suitable place. The asset details page can be one good place.

**Comment by startailcoon** [comment]
Previous discussion here #1155

**Comment by wmbutler** [comment]
Good dialog. This clearly needs UX before we move forward.

**Comment by clockworkgr** [comment]
so not gonna be able to update issuer via UI when the HF hits in 2 days?

**Comment by gibbsfromncis** [comment]
@clockworkgr I'm going to close this one asap when I come back from vacation. If somebody want to close it before I come back they are free to assign it.

**Comment by startailcoon** [comment]
Unasigned @gibbsfromncis for the moment. Free to claim by anyone

**Comment by svk31** [comment]
Gonna implement this, the UX might be temporary but at least the functionality will be there. I'm gonna move it to the Asset page but just keep the current UX for it. Then I'll split the asset page into two tabs: Info and Actions. Claiming balance pool, funding fee pool, changing owner etc will go into the Actions tab.
